### PR TITLE
dovecot: 2.3.11.3 -> 2.3.13

### PR DESCRIPTION
**Pull request by andir**
###### Motivation for this change

This fixes CVE_2020-24386, CVE-2020-25725 and a bunch of regular bugs
[1].

* CVE-2020-24386: Specially crafted command can cause IMAP hibernate to
	  allow logged in user to access other people's emails and filesystem
	  information.

* CVE-2020-25275: Mail delivery / parsing crashed when the 10 000th MIME part was
  message/rfc822 (or if parent was multipart/digest). This happened
  due to earlier MIME parsing changes for CVE-2020-12100.

[1] https://raw.githubusercontent.com/dovecot/core/2.3.13/NEWS

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))

### CODE REVIEW

**Comment by ajs124** [comment]
This seems to break the `dovecot_pigeonhole` build. There's also a new release of that.

**Comment by andir** [comment]
@ajs will test, do we have a nixos test for pigeonhole yet?

**Comment by ajs124** [comment]
We don't, but I just deployed it to production.

**Comment by andir** [comment]
@ajs124 should work now. I've also added it to the test and added the actual dovecot test to the passthru.tests attribute set. I've compiled the fts xapian plugin locally but we also do not have enough tests for it (and I do not have a working setup with that plugin enabled, right now).

File: nixos/tests/dovecot.nix [hunk 1]
```diff
@@ -4,8 +4,11 @@ import ./make-test-python.nix {
   machine = { pkgs, ... }: {
     imports = [ common/user-account.nix ];
     services.postfix.enable = true;
-    services.dovecot2.enable = true;
-    services.dovecot2.protocols = [ "imap" "pop3" ];
+    services.dovecot2= {
```
Comment (lukegb) [inline nixos/tests/dovecot.nix:7]: Nit:

```suggestion
    services.dovecot2 = {
```

**Review by lukegb — APPROVED** [review]

**Comment by lukegb** [comment]
@ofborg test opensmtpd-interaction dovecot

File: nixos/tests/dovecot.nix (hunk 1 above)
Comment (andir) [reply nixos/tests/dovecot.nix]: Fixed.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 108404` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>dovecot</li>
    <li>dovecot_fts_xapian</li>
    <li>dovecot_pigeonhole</li>
  </ul>
</details>

**Pull request merged**